tests/Contao/StringUtilTest.php

```diff
@@ -0,0 +1,236 @@
+            ],
+            'Test token replacement with special characters (#)' => [
+                'This is my ##e#mail##',
+                ['e#mail' => 'test@foobar.com'],
+                'This is my test@foobar.com',
+            ],
+            'Test comparisons (==) with regular characters (match)' => [
+                'This is my {if email==""}match{endif}',
+                ['email' => ''],
+                'This is my match',
+            ],
+            'Test comparisons (==) with regular characters (no match)' => [
+                'This is my {if email==""}match{endif}',
+                ['email' => 'test@foobar.com'],
+                'This is my ',
+            ],
+            'Test comparisons (>) with regular characters (match)' => [
+                'This is my {if value>0}match{endif}',
+                ['value' => 5],
+                'This is my match',
+            ],
+            'Test comparisons (>) with regular characters (no match)' => [
+                'This is my {if value>0}hello{endif}',
+                ['value' => -8],
+                'This is my ',
+            ],
+            'Test comparisons (<) with regular characters (match)' => [
+                'This is my {if value<0}match{endif}',
+                ['value' => -5],
+                'This is my match',
+            ],
+            'Test comparisons (<) with regular characters (no match)' => [
+                'This is my {if value<0}hello{endif}',
+                ['value' => 9],
+                'This is my ',
+            ],
+            'Test comparisons (<) with special characters (match)' => [
+                'This is my {if val&#ue<0}match{endif}',
+                ['val&#ue' => -5],
+                'This is my match',
+            ],
+            'Test comparisons (<) with special characters (no match)' => [
+                'This is my {if val&#ue<0}match{endif}',
+                ['val&#ue' => 9],
+                'This is my ',
+            ],
+            'Test whitespace in tokens not allowed and ignored' => [
+                'This is my ##dumb token## you know',
+                ['dumb token' => 'foobar'],
+                'This is my ##dumb token## you know',
+            ],
+        ];
+    }
+
+    /**
+     * Provides the data for the testParseSimpleTokens() method.
+     *
+     * @return array
+     */
+    public function parseSimpleTokensThrowsExceptionWhenContainingPhp()
+    {
+        return [
+            '(<?php)' => [
+                'This <?php var_dump() ?> is a test.',
+                false
+            ],
+            '(<?=)' => [
+                'This <?= $var ?> is a test.',
+                false
+            ],
+            '(<?)' => [
+                'This <? var_dump() ?> is a test.',
+                version_compare(PHP_VERSION, '7.0.0', '>=')
+            ],
+            '(<%)' => [
+                'This <% var_dump() ?> is a test.',
+                version_compare(PHP_VERSION, '7.0.0', '>=')
```

Comment:
Should be changed to something like `!in_array(strtolower(ini_get('asp_tags')), ['1', 'on', 'yes', 'true'])`